index.php

```diff
@@ -89,6 +103,49 @@ function viewMap () {
 	require($_SERVER['DOCUMENT_ROOT'] . TEMPLATE_PATH . "/viewMap.php");
 }
 
+function postComment() {
+	if (!isset($_GET["id"]) || !$_GET["id"]) {
+		homepage();
+		return;
+	}
+
+	require($_SERVER['DOCUMENT_ROOT'] . TEMPLATE_PATH . "/submitCommentForm.php");
+}
+
+function submitComment() {
+	if (!isset($_POST) || !$_POST || !isset($_GET["id"]) || !$_GET["id"]) {
+		homepage();
+		return;
+	}
+
+	$mapComment = new MapComment();
+	$mapComment->storeFormValues($_POST);
+	$mapComment->insert();
+}
+
+function viewComments() {
+	if (!isset($_GET["id"]) || !$_GET["id"]) {
+		homepage();
+		return;
+	}
+
+	$dispMap = Map::getByIdPublished((int) $_GET["id"], 0);
+	$results['pageTitle'] = "View Comments For " . $dispMap->name . " | CTM Map Repository";
+	require($_SERVER['DOCUMENT_ROOT'] . TEMPLATE_PATH . "/browseComments.php");
+}
+
+function reportComment() {
+	if (!isset($_GET["id"]) || !$_GET["id"] || !isset($_GET["commentId"]) || !$_GET["commentId"]) {
+		homepage();
+		return;
+	}
+
+	$mapComment = MapComment::getById($_GET["commentId"]);
+	$mapComment->flagCount++;
+	$mapComment->update();
```

Comment:
I feel like I'd want the increment to be done as an atomic operation at the database level, otherwise if 3 people report a spam comment at the same time, it could be possible for the flag count to only get incremented once due to a race condition.

Reply:
Good point. I've adjusted the increment to happen at the database level.

See commit `bd56692` for the details.